migrations/create-patient-contacts.js

```diff
@@ -0,0 +1,177 @@
+var _ = require('underscore'),
+    db = require('../db'),
+    async = require('async');
+
+var BATCH_SIZE = 100;
+
+var filterThoseWithExistingContacts = function(batch, callback) {
+  db.medic.view('medic', 'patient_by_patient_shortcode_id', {
+      keys: batch.map(function(row) {
+        return row[0];
+      })
+    }, function(err, results) {
+      if (err) {
+        return callback(err);
+      }
+
+      var existingContactShortcodes = _.pluck(results.rows, 'key');
+
+      callback(batch.filter(function(row) {
+        return !_.contains(existingContactShortcodes, row[0]);
+      }));
+    });
+};
+
+var batchCreatePatientContacts = function(batch, callback) {
+  process.stdout.write('Of ' + batch.length + ' potential patients ');
+
+  filterThoseWithExistingContacts(batch, function(filteredBatch) {
+    process.stdout.write(filteredBatch.length + ' do not have a contact. ');
+    if (filteredBatch.length === 0) {
+      process.stdout.write('\n');
+      return callback();
+    }
+
+    process.stdout.write('Getting registrations.. ');
+
+    var registrationIdsToConsider = _.flatten(_.pluck(filteredBatch, 1));
```

Comment:
Oh wait the reason I didn't do that is that it was strictly a filter i.e `A: Coll<T>, P: <Predicate> -> B: Coll<T> where B.length <= A.length` or however you want to represent that. I'll change it and maybe consider changing the function name as well.